src/logicTemplates/templates.js

```diff
@@ -0,0 +1,75 @@
+const assert = require('assert');
+const address = require("../encoding/address");
+const nacl = require('../nacl/naclWrappers');
+
+function putUvarint(buf, x){
+    let i = 0;
+    while (x > 0x80) {
+        buf.push((x&0xFF) | 0x80);
+        x >>= 7;
+        i += 1;
+    }
+    buf.push(x&0xFF);
+    return i + 1
+}
+
+const valTypes = {
+    INT : 1,
+    ADDRESS : 2,
+    BASE64 : 3
+};
+
+function inject(orig, offsets, values, valueTypes) {
+    assert.strictEqual(offsets.length, values.length);
+    assert.strictEqual(offsets.length, valueTypes.length);
+
+    let res = orig;
+
+    function replace(arr, newVal, offset, placeholderLength) {
+        let beforeReplacement = arr.slice(0, offset); // TODO ejr maybe offset + 1
+        let afterReplacement = arr.slice(offset + placeholderLength, arr.length);
+        let chunks = [beforeReplacement, Buffer.from(newVal), afterReplacement];
+        return Buffer.concat(chunks);
+    }
+
+    for (let i = 0; i < offsets.length; i++ ) {
+        let decodedLength = 0;
+        let val = values[i];
+        let valType = valueTypes[i];
+
+        switch (valType) {
+            case valTypes.INT:
+                let buf = [];
+                decodedLength = putUvarint(buf, val);
+                res = replace(res, buf, offsets[i], 1);
+                break;
+            case valTypes.ADDRESS:
+                val = address.decode(val);
+                res = replace(res, val.publicKey, offsets[i], 32);
+                break;
+            case valTypes.BASE64:
+                val = Buffer.from(val, 'base64');
+                res = replace(res, val, offsets[i], 32);
```

Comment:
I'd add a comment that the place holder of the b64 is always 32

Reply:
Also, you should take into account the length of the b64 value. it's not always going to be 32 bytes long.